File: RTD_Temperature_Controller_Angular/src/app/models/Connection.ts

//////////////////////////////////////////////////////////////////////////
/// <summary>
/// "Connection" represents a model for communication connection
/// </summary>
//////////////////////////////////////////////////////////////////////////

export class Connection{
    PortName:string
    BitsPerSecond:number
    DataBits:number
    Parity:string
    StopBits:number

    constructor(portName:string,bitsPerSecond:number,dataBits:number,parity:string,stopBits:number){
        this.PortName = portName
        this.BitsPerSecond = bitsPerSecond
        this.DataBits = dataBits
        this.Parity = parity
        this.StopBits = stopBits
    }
}

File: RTD_Temperature_Controller_DotnetAPI/Models/Connection.cs
using System.IO.Ports;

namespace RTD_Temperature_Controller_DotnetAPI.Models
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// Represents a connection configuration for communication settings.
    /// </summary>
    //////////////////////////////////////////////////////////////////////////

    public class Connection
    {
        public string PortName { get; set; }
        public int BitsPerSecond { get; set; }
        public int DataBits { get; set; }
        public Parity Parity { get; set; }
        public StopBits StopBits { get; set; }
    }
}
